### PR TITLE
Codestyle updates and simplifications

### PR DESCRIPTION
**Pull request by TheJJ**
The codestyle info files needed some updates, please complain about obscurities.

### CODE REVIEW

**Comment by LordAro** [comment]
Still disagree (reasonably strongly) with your insistence on using "not", "and" & "or"
This isn't python...

File: doc/code_style/mom.cpp [hunk 1]
```diff
@@ -17,155 +16,192 @@
...
 		// spaces here:
 		//        v  v
-		this->mass *= 2;
+		this->mass += 1;
 	}
 
 	// spaces again:
 	// v      v v  v v v     v    v
-	for (int i = 0; i < 1337; i++) {
-		this->mass += 1;
```
Comment (janisozaur) [inline doc/code_style/mom.cpp]: doesn't have line above updated

File: doc/code_style/mom.cpp [hunk 2]
```diff
@@ -17,155 +16,192 @@
...
+		}
 	}
 
-	switch (this->mass) {
-	// the case label is at the same indentation level as its switch statement:
+	switch (this->stupidity) {
+	// the case label is at the same indentation level
+	// as its switch statement:
+	// vvv
 	case 0:
```
Comment (janisozaur) [inline doc/code_style/mom.cpp]: can cases declare new variables? if so, how would blocks here look like?

**Comment by janisozaur** [comment]
how about adding a rule file for http://clang.llvm.org/docs/ClangFormat.html or some such?

**Comment by mic-e** [comment]
I agree with @LordAro; we should keep down the WTFs/minute. Any other opinions on the `not`/`and`/`or` question?

**Comment by janisozaur** [comment]
If my vote means anything, then +1 for regular (&&, ||, !) operands.

**Comment by TheJJ** [comment]
I really like the `not` `and` `or` keywords and I don't think they'll decrease the readability.
`clang-format` sounds nice, we should integrate that.

**Comment by shinyquagsire23** [comment]
I personally find && || and ! to be more readable, to be honest I didn't even know that C++ *had* a "not" "or" or "and" before this.

File: doc/code_style/mom.cpp (hunk 2 above)
Comment (TheJJ) [inline doc/code_style/mom.cpp]: `case 8:` should provide an example now.

**Comment by mic-e** [comment]
@TheJJ: Yes, `not` etc are more readable, especially when you're using Python as well as C++, but nobody in the C++ community seems to know about them, and following common practice is often the good thing. The keyword operators are bound to increase the [WTF/m](http://images.cryhavok.org/d/2471-1/WTFs+per+Minute.jpg).

**Comment by LordAro** [comment]
The thing with clang-format is that it forces you to do what it wants you to do regardless of the situation. For sure, add a make rule, but I wouldn't recommend making it part of the build system/travis. Keep it as a general guideline instead

**Comment by mic-e** [comment]
I agree with @LordAro. An auto-formatter _might_ _just_ work for Python, but C++ is far too complex, and far too quickly evolving at that. It has multiple levels of turing-completeness and there's just no way a utility will format it right.

**Comment by fr3akout** [comment]
Although the discussion about `and`, `or` and `not` has not been finished yet, the remaining codestyle updates are ok, so ready for merge. The discussion can be continued later on anyway.

**Pull request merged**